File: projects/cache/src/main/java/com/gmail/unmacaque/spring/cache/Application.java
package com.gmail.unmacaque.spring.cache;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cache.annotation.EnableCaching;

@SpringBootApplication
@EnableCaching
public class Application {
	static void main(String[] args) {
		SpringApplication.run(Application.class, args);
	}
}

File: spring-kotlin/src/main/kotlin/com/gmail/unmacaque/spring/Application.kt
package com.gmail.unmacaque.spring

import com.gmail.unmacaque.spring.domain.Message
import com.gmail.unmacaque.spring.domain.MessageRepository
import org.springframework.boot.ApplicationRunner
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import java.util.stream.StreamSupport

@SpringBootApplication
class Application {

    @Bean
    fun init(messageRepository: MessageRepository) = ApplicationRunner {
        messageRepository.save(Message("Homer", "Marge", "I like beer."))
        messageRepository.save(Message("Bart", "everyone", "Eat my shorts!"))
        messageRepository.save(Message("Homer", "Bart", "Why you little…"))

        StreamSupport
                .stream(messageRepository.findAll().spliterator(), false)
                .forEach(::println)
    }

}

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}
